src/app.ts

```diff
@@ -164,28 +184,27 @@ const getGenerationMetrics = (
   origAttributes: Set<string>,
   genAttributes: Set<string>,
   inputWords: Set<string>
-): GenerationMetrics => {
-  const isTitleChanged = origTitle !== genTitle;
-  const attributesAdded = genAttributes.size > origAttributes.size;
-  const newWordsAdded = genTitle.split(' ').some((genTitleWord: string) => {
-    return !inputWords.has(genTitleWord);
-  });
-  const genAttributeValuesAdded = true;
-  return {
-    attributesAreAdded: attributesAdded,
-    generatedValuesAdded: genAttributeValuesAdded,
-    titleChanged: isTitleChanged,
-    newWordsAdded: newWordsAdded,
-    scores: () => {
-      return [
-        (Number(attributesAdded) +
-          Number(genAttributeValuesAdded) +
-          Number(isTitleChanged) +
-          Number(newWordsAdded)) /
-          4,
-      ];
-    },
-  };
+): string[] => {
+  const titleChanged = origTitle !== genTitle;
+  const addedAttributes = Util.getSetDifference(genAttributes, origAttributes);
+  const newWordsAdded = new Set<String>();
+  const genTitleWords = genTitle.match(WORD_MATCH_REGEX);
+  if (genTitleWords !== null) {
+    genTitleWords
+      .filter((genTitleWord: string) => !inputWords.has(genTitleWord))
+      .forEach((newWord: string) => newWordsAdded.add(newWord));
+  }
+  const totalScore =
+    (Number(addedAttributes.length > 0) +
+      Number(titleChanged) +
+      Number(newWordsAdded.size === 0)) /
+    3;
+  return [
+    totalScore.toString(), // 0-1 score total
+    titleChanged.toString(), // Title Changed
+    addedAttributes.map((attr: string) => `<${attr}>`).join(' '), // Attributes added
+    [...newWordsAdded].join(SEPARATOR), //hallucinated words added to title
```

Comment:
[...newWordsAdded].join(` ${SEPARATOR} `), // added leading and trailing space for better readability in Sheets